waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/server/FederatedHMSHandler.java

```diff
@@ -1266,14 +1274,14 @@ private DatabaseMapping checkWritePermissionsForPrivileges(PrivilegeBag privileg
   @Override
   @Loggable(value = Loggable.DEBUG, skipResult = true, name = INVOCATION_LOG_NAME)
   public List<String> set_ugi(String user_name, List<String> group_names) throws MetaException, TException {
-    List<DatabaseMapping> mappings = databaseMappingService.getDatabaseMappings();
+    List<DatabaseMapping> mappings = databaseMappingService.getAllDatabaseMappings();
```

Comment:
Mostly whitespace in this class, but this is key, it gets all metastores and does the call as we will cache it and not actually call the underlying metastore we don't care if the are unavailable and it prevent availability checks.